library/src/scala/runtime/JavaRecordMirror.scala

```diff
@@ -0,0 +1,34 @@
+package scala.runtime
+
+import java.lang.Record
+import java.lang.reflect.Constructor
+import scala.reflect.ClassTag
+
+// TODO: Rename to JavaRecordReflectMirror
+object JavaRecordMirror:
```

Comment:
Is this going to be a big deal? It means that compiling the scala library would need JDK 16+, but having this on the classpath and not loaded should not be an issue?